### PR TITLE
graphql_ppx 1.0

### PR DESCRIPTION
**Pull request by baransu**
🚧🚧 Work in Progress! 🚧🚧

For making this work with Apollo please check:
https://github.com/Astrocoders/reason-apollo-hooks/pull/117. (1.0 adds a lot of stuff that makes the client incompatible with the current version, we'll try to keep the apollo PR up to date with the latest version of this PR).

Features:
(edited by @jfrolich)

- [x] compile to records by default by @jfrolich (#67)
- [x] tagged template literal syntax support (#59)
- [x] object switch to compile to objects
- [x] generate raw types in `Query.Raw.t` (#98)
- [x] improve compatibility with gentype (#104)
- [x] serialize function (#101)
- [x] graphql configuration in `bsconfig.json` (#40)
- [x] default custom scalars in configuration file
- [x] better errors when a field is missing (#105)
- [x] better errors when a fragment is missing (#106)
- [x] fix fragment spread issues (#107)
- [x] Better editor support (#118)
- [x] Add integration tests
- [ ] refuse to serialize `@bsVariant`
- [ ] rename `@bsVariant` to ppxVariant (make both work) 
- [ ] Generate documentation in doc comments for better editor DX
- [x] New API
- [x] docusaurus (or similar) documentation site (#102)

We are almost there:

🚀 Launch checklist 🚀:
- [ ] freeze features
- [ ] contribute compatible version of `reason-apollo-hooks` (in progress: https://github.com/reasonml-community/reason-apollo-hooks/pull/117)
- [ ] contribute compatible version of `reason-urql`
- [ ] contribute compatible version of `@jfrolich/reason-gatsby`
- Critical documentation
  - [ ] README.md
  - [ ] API docs 
  - [ ] getting started
  - [ ] installation

### CODE REVIEW

**Comment by BlueHotDog** [comment]
Wow. this looks such an awesome release, i'm just wondering if we can go with 1.0 without everything so people can start exploring this major change? maybe documentation can come at a later phase?

**Comment by jsiebern** [comment]
@BlueHotDog You can already use 1.0! Just install `@baransu/graphql_ppx_re@next` to get the current stable release of 1.0. You can also install `@dev` to get the latest in the 1.0 branch that is not yet released.

And use this PR over at `reason-apollo-hooks`:
https://github.com/Astrocoders/reason-apollo-hooks/pull/117

**Comment by BlueHotDog** [comment]
Thanks, and i actually did try it(and it was amazing!), but it seems like no URQL support.
is there any work being done on that front?
any documentation of what needs to be done?

Thanks,
Danni Friedland

**Comment by jfrolich** [comment]
> Thanks, and i actually did try it(and it was amazing!), but it seems like no URQL support. is there any work being done on that front? any documentation of what needs to be done? Thanks, Danni Friedland
> […](#)
> On May 15, 2020, 23:44 +0300, Jonathan ***@***.***>, wrote: @BlueHotDog You can already use 1.0! Just install @***@***.*** to get the current stable release of 1.0. You can also install @dev to get the latest in the 1.0 branch that is not yet released. And use this PR over at reason-apollo-hooks: [reasonml-community/reason-apollo-hooks#117](https://github.com/reasonml-community/reason-apollo-hooks/pull/117) — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or unsubscribe.

That is right. I'd like to contribute this to `reason-urql` soon. But we have one breaking API change that is going to land (final API) #138.

**Comment by BlueHotDog** [comment]
I would love to help push this as much as possible.
Is there a place documenting all the breaking changes?

**Comment by jfrolich** [comment]
> I would love to help push this as much as possible.
> Is there a place documenting all the breaking changes?

Not yet. Mostly it's object => records. `make` function is now deprecated, you can use `makeVariables` and `self` (not yet merged in 1.0, but that is the new `definition`).

File: src/base/result_decoder.re [hunk 1]
```diff
@@ -396,70 +501,147 @@ and unify_field = (error_marker, config, field_span, ty) => {
...
         loc,
-        Res_custom_decoder(config.map_loc(span), fn_name, parser_expr),
-      )
+        loc_key,
+        type_:
+          make_error(
+            error_marker,
+            config.map_loc,
+            span,
+            "pxxDecoder must be given 'module' argument",
```
Comment (anmonteiro) [inline src/base/result_decoder.re:534]:
```suggestion
            "ppxDecoder must be given 'module' argument",
```

**Pull request merged**